### PR TITLE
chore: [CDS-69491]: digest field enhancements

### PR DESCRIPTION
**Pull request by namanharness**
### Summary

Following enhancements need to be done for digest field (**Docker ,GAR,GCR,ECR and ACR** artifact source) 
1. Disable digest if version/versionRegex is marked as fixed value and not provided any value - Done for services & templates
2. When required fields are runtime or not provided value, show that these fields are required for fetching digest. - Done for Services & templates page

Both of above handlings are similar to version.
When user selects tagRegex: No API call for listing digest, User can provide value. - done , we use string text input component to make this feasible - inside and outisde pipeline studio - Docker

1. Tooltip for digest - done
2. Mark field as Optional - done
3. Make placeholder 'Enter digest' similar to version. - done

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ACRArtifact/AcrDigestField.tsx [hunk 1]
```diff
@@ -54,6 +60,32 @@ export function AcrArtifactDigestField({
     get(formik?.values, 'tag')
   )
 
+  const tagValue = defaultTo(formik?.values?.tag?.value, formik?.values?.tag)
+  const digestValue = defaultTo(formik?.values?.digest?.value, formik?.values?.digest)
+
+  const isDigestDisabled = React.useCallback(() => {
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ACRArtifact/AcrDigestField.tsx:66]: Please use useMemo here

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/DockerRegistryArtifact/DockerArtifactDigestField.tsx [hunk 2]
```diff
@@ -44,8 +54,32 @@ export function DockerArtifactDigestField({
     formikDigestValueField: formik?.values?.digest
   }
 
-  const canFetchDockerDigest = canFetchDigest(formik?.values?.imagePath, formik?.values?.tag, connectorRefValue)
+  const tagValue = defaultTo(formik?.values?.tag?.value, formik?.values?.tag)
+  const digestValue = defaultTo(formik?.values?.digest?.value, formik?.values?.digest)
 
+  const isDigestDisabled = React.useCallback(() => {
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/DockerRegistryArtifact/DockerArtifactDigestField.tsx:60]: same

File: src/modules/70-pipeline/utils/stageHelpers.ts [hunk 3]
```diff
@@ -200,6 +200,23 @@ export const getHelperTextString = (
   }`
 }
 
+export const getHelperTextStringForDigest = (
+  invalidFields: string[],
+  getString: (key: StringKeys) => string,
+  isServerlessDeploymentTypeSelected = false,
+  defaultErrorMessage = ''
+): string => {
+  return `${invalidFields.length > 1 ? invalidFields.join(', ') : invalidFields[0]} ${
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/utils/stageHelpers.ts:209]: Lets extract it out in a variable and use it at both all places

const areMultipleInvalidFieldsPresent = invalidFields.length > 1

**Review by samarth-sharma31 — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

**Pull request merged**